### ISSUE TITLE
indexing should permit uint as well as int

### ISSUE THREAD

**Issue by joshmoore** [PyTables/PyTables#43]
I have a column of type uint32 in one table containing values for row-numbers in another table. Currently, I am forced to cast this to int32 before I can use it as an index. I think that makes no sense, logically, so I set the type of this issue to defect rather than enhancement.

```python
In [1]: import tables

In [3]: t = tables.openFile('tutorial1.h5')

In [8]: t.root.detector.readout[np.array([2, 3, 4], dtype=np.int32)]
Out[8]: 
array([(512, 2, 256.0, 2, 8, 34359738368, 'Particle:      2', 4.0),
       (768, 3, 6561.0, 3, 7, 51539607552, 'Particle:      3', 9.0),
       (1024, 4, 65536.0, 4, 6, 68719476736, 'Particle:      4', 16.0)], 
      dtype=[('ADCcount', '<u2'), ('TDCcount', '|u1'), ('energy', '<f8'), ('grid_i', '<i4'), ('grid_j', '<i4'), ('idnumber', '<i8'), ('name', '|S16'), ('pressure', '<f4')])

In [9]: t.root.detector.readout[np.array([2, 3, 4], dtype=np.uint32)]
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)

/storage4/home/gerrit/checkouts/pytables-2.2/examples/<ipython console> in <module>()

/storage4/home/gerrit/.local/lib/python2.6/site-packages/tables/table.pyc in __getitem__(self, key)
   1709         # Try with a boolean or point selection

   1710         elif type(key) in (list, tuple) or isinstance(key, numpy.ndarray):
-> 1711             coords = self._pointSelection(key)
   1712             return self._readCoordinates(coords, None)
   1713         else:

/storage4/home/gerrit/.local/lib/python2.6/site-packages/tables/leaf.pyc in _pointSelection(self, key)
    587                 coords = numpy.asarray(key, dtype="i8")
    588         else:
--> 589             raise TypeError("Only integer coordinates allowed.")
    590         # We absolutely need a contiguous array

    591         if not coords.flags.contiguous:

TypeError: Only integer coordinates allowed.
```

It's probably a very easy fix, I might have a look at it this weekend.

**Comment by 153957** [comment]
This issues has been fixed at some point. Using an array as index works with dtype=uint32, and even dtype=uint64.

However, using a single uint64 value does not work, see #475.

**Comment by andreabedini** [comment]
Should be fixed by #496.
